src/hotspot/share/gc/g1/g1ThreadLocalData.hpp

```diff
@@ -36,6 +36,15 @@
 class G1ThreadLocalData {
 private:
   SATBMarkQueue _satb_mark_queue;
+  // The current base address of the card table. Accessed by the barrier to do
+  // the card mark. Changed as required by the refinement control thread to
+  // implement card table switching.
+  //
+  // Tests showed that embedding this value in the TLS block is the cheapest
+  // way for fast access to this value in the barrier.
+  // E.g. embedding an address to that value directly into the code stream similar
+  // to Serial/Parallel and then loading from that was found to be slower at least
```

Comment:
Not sure if I misunderstood this text or not, but Serial/Parallel doesn't embed the "address" into the code stream -- they embed the actual value in the code stream, since the value (card-tabe-base) is constant. G1 can't/doesn't do that, since card-table-base is not constant.

Reply:
Similar also does not mean equal.

Reply:
maybe just removing the "an address" to avoid the confusion?